File: src/main/java/kr/hyfata/zero/ZeroModules.java
package kr.hyfata.zero;

import kr.hyfata.zero.modules.mailbox.ZeroMailbox;
import kr.hyfata.zero.modules.jangsa.ZeroJangsa;
import kr.hyfata.zero.modules.scoreboard.ZeroScoreBoard;
import org.bukkit.plugin.java.JavaPlugin;

public class ZeroModules {
    ZeroScoreBoard zeroScoreBoard;
    ZeroMailbox zeroMailbox;
    ZeroJangsa zeroJangsa;

    public ZeroModules(JavaPlugin plugin) {
        zeroScoreBoard = new ZeroScoreBoard(plugin);
        zeroMailbox = new ZeroMailbox(plugin);
        zeroJangsa = new ZeroJangsa(plugin);
    }

    public void onDisable() {
        zeroScoreBoard.onDisable();
        zeroMailbox.onDisable();
    }

    public ZeroScoreBoard getZeroScoreBoard() {
        return zeroScoreBoard;
    }

    public ZeroMailbox getZeroMailboxUI() {
        return zeroMailbox;
    }

    public ZeroJangsa getZeroJangsa() {
        return zeroJangsa;
    }
}

File: src/main/java/kr/hyfata/zero/ZeroModules.kt
package kr.hyfata.zero

import kr.hyfata.zero.modules.actionbar.ZeroActionbar
import kr.hyfata.zero.modules.mailbox.ZeroMailbox
import kr.hyfata.zero.modules.scoreboard.ZeroScoreBoard
import org.bukkit.plugin.java.JavaPlugin

class ZeroModules(plugin: JavaPlugin) {
    var zeroScoreBoard: ZeroScoreBoard = ZeroScoreBoard(plugin)
    var zeroMailbox: ZeroMailbox = ZeroMailbox(plugin)
    var zeroActionbar: ZeroActionbar = ZeroActionbar(plugin)

    fun onDisable() {
        zeroScoreBoard.onDisable()
        zeroMailbox.onDisable()
        zeroActionbar.onDisable()
    }
}
